### DISCUSSION TITLE
Clarification on dependabot.yaml and code security settings

### DISCUSSION THREAD

**Discussion by piotrekkr** [community/community#146660] [Code Security] [answered]
### Select Topic Area

Question

### Body

Hello. Looking at docs and it is not clear to me if I need to enable dependabot updates and grouping in repo/org setting or I can simply just create `dependabot.yaml` file and it will work out of the box?
If I have this in org configuration:

![image](https://github.com/user-attachments/assets/bed456c5-cbb1-45b1-aedc-f5656ca1c09b)

And this in repo configuration

![image](https://github.com/user-attachments/assets/b2b4e76b-58b6-429f-8870-b635cc5d53cf)

Do I need to click `Enable` for `Dependabot security updates` and ` Grouped security updates` in each repository that contains `dependabot.yaml`? Will just `dependabot.yaml` be enough? Thank you.

**Comment by carogalvin** [comment, marked as answer]
@piotrekkr it depends if you want Dependabot PRs that provide updates that resolve alerts (i.e. "security updates") or PRs that update your dependencies to the latest version (i.e. "version updates"). They are enabled separately. For security updates, check the box here and it'll work without dependabot.yml. For version updates, you must check in a dependabot.yml.

**Reply by piotrekkr** [reply]
Thanks for reply. What happens when I have both, `dependabot.yaml` and settings enabled? `dependabot.yaml` takes precedence over settings in repo?

**Reply by carogalvin** [reply]
Then you will get both security & version updates, and your configuration from the YAML will apply. For more information on how to set up the YAML to configure version and/or security updates, you can check out our docs: https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file

**Reply by jesseduffield** [reply]
@carogalvin I'm still confused by this. If I have the dependabot.yml file checked in, and I only enable security updates in the settings, does that mean that the file will only apply to security updates, and no version updates will happen?

I'm asking because after adding the dependabot.yml file, the only way to disable version updates was to delete the file, given that there was no 'disable' button in the settings against the version updates option; only a 'configure' button.

**Reply by carogalvin** [reply]
Adding a dependabot.yml will automatically turn on version updates. If you want security updates and not version updates, you can set `open-pull-requests-limit` to 0: https://docs.github.com/en/code-security/dependabot/working-with-dependabot/dependabot-options-reference#open-pull-requests-limit-
